pkg/cloudprovider/vsphere/instances.go

```diff
@@ -130,14 +130,20 @@ func (i *instances) InstanceID(ctx context.Context, nodeName types.NodeName) (st
 // InstanceType returns the type of the instance identified by name.
 func (i *instances) InstanceType(ctx context.Context, name types.NodeName) (string, error) {
 	klog.V(4).Info("instances.InstanceType() called")
-	return i.nodeManager.nodeNameMap[string(name)].NodeType, nil
+	if nodeInfo, ok := i.nodeManager.nodeNameMap[string(name)]; ok {
+		return nodeInfo.NodeType, nil
+	}
+	return "", nil
```

Comment:
Is it appropriate to return a nil error here or should we error if the object doesn't exist in cache?